hmpps-prisoner-search/src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/search/services/ReferenceDataService.kt

```diff
@@ -18,29 +19,29 @@ class ReferenceDataService(
   @Value("\${search.detailed.max-results}") private val maxSearchResults: Int = 200,
   @Value("\${search.detailed.timeout-seconds}") private val searchTimeoutSeconds: Long = 10L,
 ) {
+  // we have implemented a cache loader in CacheConfig so that will call the findReferenceData method below
+  @Cacheable(cacheNames = ["referenceData"], key = "#attribute")
+  fun findReferenceDataCached(attribute: ReferenceDataAttribute) = ReferenceDataResponse()
+
   fun findReferenceData(attribute: ReferenceDataAttribute): ReferenceDataResponse {
     val searchSourceBuilder = createSourceBuilder(attribute)
     val searchRequest = SearchRequest(arrayOf(OpenSearchIndexConfiguration.PRISONER_INDEX), searchSourceBuilder)
 
-    return try {
-      val searchResponse = elasticsearchClient.search(searchRequest)
-      val aggregation: MultiBucketsAggregation = searchResponse.aggregations.get(attribute.name)
-      ReferenceDataResponse(
-        aggregation.buckets.map {
-          ReferenceData(it.keyAsString)
-        }.sortedBy { it.key },
-      )
-    } catch (e: Throwable) {
```

Comment:
this looked wrong - don't like the idea of catching all throwables and then just returning no results back to the client